src/bin/ion/commands/beta/to/json.rs

```diff
@@ -57,75 +41,44 @@ pub fn run(_command_name: &str, matches: &ArgMatches) -> Result<()> {
                 output_file
             )
         })?;
-        Box::new(file)
+        Box::new(BufWriter::new(file))
     } else {
         Box::new(stdout().lock())
     };
 
-    if let Some(input_file_iter) = matches.get_many::<String>("input") {
-        for input_file in input_file_iter {
-            let mut file = File::open(input_file)
-                .with_context(|| format!("Could not open file '{}'", input_file))?;
-            convert(&mut file, &mut output, format)?;
+    if let Some(input_file_names) = matches.get_many::<String>("input") {
+        // Input files were specified, run the converter on each of them in turn
+        for input_file in input_file_names {
```

Comment:
We see this pattern repeated over and over again in `ion-cli`- in the long term it would be great if we had a common solution to it. Nothing to do here at the moment.